File: src/main/java/com/gtladd/gtladditions/api/machine/trait/IWirelessNetworkEnergyHandler.java
package com.gtladd.gtladditions.api.machine.trait;

import java.math.BigInteger;

public interface IWirelessNetworkEnergyHandler {

    boolean consumeEnergy(int energy);

    boolean consumeEnergy(long energy);

    boolean consumeEnergy(BigInteger energy);

    BigInteger getMaxAvailableEnergy();

    boolean isOnline();
}

File: src/main/java/com/gtladd/gtladditions/api/machine/trait/IWirelessNetworkEnergyHandler.kt
package com.gtladd.gtladditions.api.machine.trait

import java.math.BigInteger

interface IWirelessNetworkEnergyHandler {
    fun consumeEnergy(energy: Int): Boolean

    fun consumeEnergy(energy: Long): Boolean

    fun consumeEnergy(energy: BigInteger): Boolean

    val maxAvailableEnergy: BigInteger

    val isOnline: Boolean
}
